siobrultech_protocols/gem/protocol.py

```diff
@@ -15,16 +16,58 @@
 LOG = logging.getLogger(__name__)
 
 PACKET_HEADER = bytes.fromhex("feff")
+API_RESPONSE_WAIT_TIME_SECONDS = 3.0  # Time to wait for an API response
+PACKET_DELAY_CLEAR_TIME_SECONDS = 3.0  # Time to wait after a PDL request so that GEM can finish sending any pending packets
 
 
 class PacketProtocol(asyncio.Protocol):
```

Comment:
I wonder if we should rename this now, or maybe still provide the simple, basic one in addition to this?

Reply:
I'm up for it. Name preference?

Reply:
Well, do you think we should split and inherit, or just have an uber protocol?

Argument for pro-splitting would be that we might be able to make a packet protocol that decode data for both the GEM and the ECM.  Argument against is that it complicates stuff.

As for name, maybe something like `GEMProtocol`?  I'm not thrilled with it though, so open to ideas.

Reply:
I split out a `BidirectionalProtocol` subclass. It's accessing some of the parent class's nominally-private fields and methods, which is a little dirty, but overall I like the split. It would be easy to implement it without doing that, though; it just saves a little duplication that way.